tests/src/main/java/org/hibernate/beanvalidation/tck/tests/constraints/containerelement/ContainerElementConstraintOptionalTest.java

```diff
@@ -0,0 +1,221 @@
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.method( "returnStringOptional" )
+								.returnValue()
+						)
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "c")
+	public void method_parameter_constraint_provided_as_type_parameter_of_an_optional_gets_validated()
+			throws Exception {
+		Method method = TypeWithOptional5.class.getDeclaredMethod( "setValues", Optional.class );
+		Object[] values = new Object[] { Optional.of( "" ) };
+
+		Set<ConstraintViolation<TypeWithOptional5>> constraintViolations = getValidator().forExecutables().validateParameters(
+				new TypeWithOptional5(),
+				method,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.method( "setValues" )
+								.parameter( "optionalParameter", 0 )
+						)
+		);
+	}
+
+	@Test
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "a")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "b")
+	@SpecAssertion(section = Sections.CONSTRAINTDECLARATIONVALIDATIONPROCESS_CONTAINERELEMENTCONSTRAINTS, id = "c")
+	public void constructor_parameter_constraint_provided_on_type_parameter_of_an_optional_gets_validated()
+			throws Exception {
+		Constructor<TypeWithOptional6> constructor = TypeWithOptional6.class.getDeclaredConstructor( Optional.class );
+		Object[] values = new Object[] { Optional.of( "" ) };
+
+		Set<ConstraintViolation<TypeWithOptional6>> constraintViolations = getValidator().forExecutables().validateConstructorParameters(
+				constructor,
+				values
+		);
+
+		assertThat( constraintViolations ).containsOnlyViolations(
+				violationWith()
+						.constraintType( NotBlank.class )
+						.propertyPath( pathWith()
+								.constructor( TypeWithOptional6.class )
+								.parameter( "optionalParameter", 0 )
+						)
+		);
+	}
+
+	private static class TypeWithOptional1 {
+
+		@SuppressWarnings("unused")
+		private Optional<@NotBlank String> stringOptional;
+	}
+
+	private static class TypeWithOptional2 {
+
+		@NotNull
+		private Optional<@NotBlank String> stringOptional;
+	}
+
+	private static class TypeWithOptional3 {
+
+		private Optional<String> stringOptional;
+
+		@SuppressWarnings("unused")
+		public Optional<@NotNull @NotBlank String> getStringOptional() {
+			return stringOptional;
+		}
+	}
+
+	private static class TypeWithOptional4 {
+
+		private Optional<String> stringOptional;
+
+		@SuppressWarnings("unused")
+		public Optional<@NotNull @NotBlank String> returnStringOptional() {
```

Comment:
An optional of not null is a bit painful. Use another constraint perhaps? Do we have one test on the payload business btw. (applying it to the wrapper)?